src/Command/AllCommand.php

```diff
@@ -82,13 +95,10 @@ public function execute(Arguments $args, ConsoleIo $io): ?int
             $tables = [$name];
         }
 
-        $commands = [
-            new ModelCommand(),
-            new ControllerCommand(),
-            new TemplateCommand(),
-        ];
-        foreach ($tables as $table) {
-            foreach ($commands as $command) {
+        foreach ($this->commands as $commandName) {
```

Comment:
You are changing the order from tables => commands to commands => tables
But I think this is actually correct.
As you first need all the models, before you can do the controller and template (to have the relations). :+1:

Reply:
Tables are instantiated earlier on in the table scanner or how were the commands previously aware of relations? 

Also not so confident that changing the order is a good thing (BC?) but am not very familiar with the intricacies of bake either.

Reply:
Just saying.
A template task might need to be more aware of all the tables existing
Thus letting the MVC arrow guide us here to finish each per type (command) rather than class could definitely make it less error prone.

Reply:
In either case, commands are already aware of all tables which are instantiated earlier is what am trying to say.

What concerns me about the order is more BC than anything else.

Reply:
@markstory Are you fine with this change?
I think the new order does make sense.

Reply:
@dereuromark I fail to understand the logic behind switching the order. 

Aside from running into BC risk, what else does it "add"?

Reply:
Proper order of layers first, before moving to the next layer.
It makes so much for sense then doing this per model first, possibly needing to redo e.g. templating and controller based on some changed model, which often escalates into multiple ones due to relation changes.
So making this layer=>model is more logical than model=>layer

Reply:
> Proper order of layers first, before moving to the next layer.

When I bake, the right order is by type of "resource", specially in CRUD type of applications where I use bake the most. Which is also why I always bake my tables before baking my controllers. If you aren't using `bake all`, in what order do you bake? 
 
> possibly needing to redo e.g. templating and controller based on some changed model, which often escalates into multiple ones due to relation changes

How does switching the order minimize the "redo"-type of work when model(s) change? Or did I misunderstand the "advantage" you are trying to explain?

Reply:
> minimize the "redo"-type of work when model(s) change?

If it breaks in the controller, you know you are done with model and can also just do controllers now.

This was meant for "all" command mainly. Which is exactly the class we are talking about here.